staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -169,6 +169,10 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 
 // Encode serializes the provided object to the given writer.
 func (s *Serializer) Encode(obj runtime.Object, w io.Writer) error {
+	if cvo, ok := obj.(*runtime.CachingVersionedObject); ok {
```

Comment:
Maybe make a "CustomSerializer" interface and check for that, so you don't have to build any new type into this repo?

Reply:
@lavalamp - so to clarify that I understand correctly, what you suggest is:
1. Create an interface that would be part of pkg/runtime
2. Check if my object (defined outside pkg/runtime) implements this interface and if so call appropriate method.
Is that what you meant?

Reply:
OK - so I'v been thinking about it and have an idea how to make it in a clear way. I will try to find some time to experiment with it a bit more and will get back to you. Though it may take some time until I will find this time :)